src/main/scala/org/fortysevendeg/sparksftp/common/HiveUserData.scala

```diff
@@ -39,9 +44,29 @@ object HiveUserData {
     salariesDataFromHive.show(false)
 
     //Excerpt from the joined table
-    sparkSession.sql("select name,salary from user_salary").show(false)
+    val userSalaries = sparkSession.sql("select name,salary from user_salary")
+    userSalaries.show(false)
+
+    HiveUserData(userDataFromHive, salariesDataFromHive, userSalaries)
+  }
+
+  def persistAndReadUserData(
+      sparkSession: SparkSession,
+      users: DataFrame,
+      salaries: DataFrame
+  ): IO[HiveUserData] = IO {
+    HiveUserData.persistUserData(sparkSession, users, salaries)
+    HiveUserData.readUserData(sparkSession)
```

Comment:
These two operations look like both might return different _IO effects_, so we might want to modify them and modify this piece of code to have a for-comp.

Reply:
Good catch, changed in 025bf8f